subsys/shell/shell_log_backend.c

```diff
@@ -136,6 +136,18 @@ void z_shell_log_backend_disable(const struct shell_log_backend *backend)
 	backend->control_block->state = SHELL_LOG_BACKEND_DISABLED;
 }
 
+void z_shell_log_backend_resume(const struct shell_log_backend *backend)
+{
+	if (backend->control_block->state != SHELL_LOG_BACKEND_ENABLED) {
+		uint32_t level =
+			(CONFIG_SHELL_BACKEND_SERIAL_LOG_LEVEL > LOG_LEVEL_DBG) ?
```

Comment:
Is there any more generic replacement for `CONFIG_SHELL_BACKEND_SERIAL_LOG_LEVEL` ?

Reply:
@de-nordic

Reply:
The best solution, I can see for this, would be to save previous log level.
This means that we should probably modify `log_backend_disable` to return previous level, or define `log_backend_suspend` that would do that, and then use the level with `z_shell_log_backend_resume`.

Reply:
But this also will store only one log level, instead of "per log source level", right ?

Reply:
Do you know why `log_backend_disable` deactivate the backend and also change the loglevel(s) ?
I would expect that `log_backend_suspend` implementation will be to just deactivate the backend without changing the loglevel(s)

Reply:
Unfortunately I currently posses  "here be dragons" level of knowledge on internals of the log subsystem.

Reply:
I found a way simpler solution to this problem, I will update this PR soon.

Reply:
I updated the PR.
The solution is to just change the backend state `backend->control_block->state` between enabled/disabled.
This also will maintain the log level changes in the runtime.

Reply:
I haven't looked into this closely, but it seems you're suspending logging before transmitting an SMP packet, but you only resume it on reception of a non-SMP byte.

Shouldn't there be a resume when the packet has been transmitted completely? Otherwise logging will stay disabled until we receive something.

Reply:
@ExtremeGTX  the mcumgr does have callback mechanism in img_mgmt; callbacks are registered with: `img_mgmt_register_callbacks`

Reply:
@ExtremeGTX Then propose new callbacks.